libs/langchain/langchain/vectorstores/redis/redis.py

```diff
@@ -308,65 +380,106 @@ def similarity_search_limit_score(
             an empty list is returned.
 
         """
-        docs_and_scores = self.similarity_search_with_score(query, k=k)
-        return [doc for doc, score in docs_and_scores if score < score_threshold]
+        # TODO check for correct redisearch version
+        docs = self._similarity_search(
+            query,
+            k=k,
+            score_threshold=score_threshold,
+            filter=filter,
+            return_scores=False,
+        )
+        return docs  # type: ignore
+
+    def similarity_search_with_score(
+        self,
+        query: str,
+        k: int = 4,
+        filter: Optional[RedisFilter] = None,
+        **kwargs: Any,
+    ) -> List[Tuple[Document, float]]:
+        """Run similarity search with distance."""
+        docs = self._similarity_search(query, k=k, filter=filter, return_scores=True)
+        return docs  # type: ignore
```

Comment:
to avoid having to ignore mypy, could just move all of _similarity_search into similarity_search_with_score, return docs and scores, and then similarity_search looks like
```python
docs_and_scores = self.similarity_search_with_scores(...)
return [d for d, _ in docs_and_scores]
```

Reply:
So introduce a loop just to satisfy mypy? This and code duplication was my issue with the last approach.